tools/appveyor.psm1

```diff
@@ -0,0 +1,201 @@
+        {
+            if($retryCount -gt 0)
+            {
+                Write-Verbose "Retrying updating test artifacts..."
+            }
+
+            $retryCount++
+            $resolvedResultsPath = (Resolve-Path $resultsFile)
+            try {
+                (New-Object 'System.Net.WebClient').UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", $resolvedResultsPath)
+                $pushedResults = $true
+            }
+            catch {
+                Write-Warning "Pushing test result failed..."
+            }
+
+            try {
+                Push-AppveyorArtifact $resolvedResultsPath
+                $pushedArtifacts = $true
+            }
+            catch {
+                Write-Warning "Pushing test Artifact failed..."
+            }
+        }
+
+        if(!$pushedResults -or !$pushedResults)
+        {
+            Write-Warning "Failed to push all artifacts for $resultsFile"
+        }
+    }
+    else 
+    {
+        Write-Warning "Not running in appveyor, skipping upload of test results: $resultsFile"
+    }
+}
+
+# Implement AppVeyor 'Test_script'
+function Invoke-AppVeyorTest 
+{
+    #
+    # CoreCLR
+    $env:CoreOutput = Split-Path -Parent (Get-PSOutput -Options (New-PSOptions -Publish -Configuration $buildConfiguration))
+    Write-Host -Foreground Green 'Run CoreCLR tests'
+    $testResultsFile = "$pwd\TestsResults.xml"
+    if(!(Test-Path "$env:CoreOutput\powershell.exe"))
+    {
+        throw "CoreCL PowerShell.exe was not built"
+    }
+    
+    & ("$env:CoreOutput\powershell.exe") -noprofile -noninteractive -c "Set-ExecutionPolicy -Scope Process Unrestricted; Invoke-Pester test/powershell -Tag 'CI' -ExcludeTag 'Slow' -OutputFormat NUnitXml -OutputFile $testResultsFile"
+    Update-AppVeyorTestResults -resultsFile $testResultsFile
+    
+    #
+    # FullCLR
+    $env:FullOutput = Split-Path -Parent (Get-PSOutput -Options (New-PSOptions -FullCLR))
+    Write-Host -Foreground Green 'Run FullCLR tests'
+    $testResultsFileFullCLR = "$pwd\TestsResults.FullCLR.xml"
+    Start-DevPowerShell -FullCLR -NoNewWindow -ArgumentList '-noprofile', '-noninteractive' -Command "Set-ExecutionPolicy -Scope Process Unrestricted; Invoke-Pester test/fullCLR -ExcludeTag 'Slow' -OutputFormat NUnitXml -OutputFile $testResultsFileFullCLR"
+    Update-AppVeyorTestResults -resultsFile $testResultsFileFullCLR
+ 
+
+    #
+    # Fail the build, if tests failed
+    Write-Host -Foreground Green 'Upload CoreCLR test results'
+    $x = [xml](cat -raw $testResultsFile)
+    if ([int]$x.'test-results'.failures -gt 0)
+    {
+    throw "$($x.'test-results'.failures) tests in test/powershell failed"
+    }
+
+    Write-Host -Foreground Green 'Upload FullCLR test results'
+    $x = [xml](cat -raw $testResultsFileFullCLR)
+
+    if ([int]$x.'test-results'.failures -gt 0)
+    {
+        throw "$($x.'test-results'.failures) tests in test/fullCLR failed"
```

Comment:
I don't see a difference

---
In reply to: [76693531](https://github.com/PowerShell/PowerShell/pull/2121#discussion_r76693531) [](ancestors = 76693531)